.github/workflows/ci.yml

```diff
@@ -0,0 +1,70 @@
+name: CI
+on:
+  push:
+    branches: [main, feat/setup-ci-cd]
+  pull_request:
+    branches: [main]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+      - run: npm ci
+      - run: cargo build --verbose
+      - run: cargo test --verbose
+
+  docker-build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: read
+    env:
+      REGISTRY: ghcr.io
+    steps:
+      - uses: actions/checkout@v4
+      - uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build Docker image
+        uses: docker/build-push-action@v6
+        with:
+          context: .
+          cache-from: type=registry,ref=ghcr.io/filecoin-station/fil-deal-ingester
+          cache-to: type=inline
+
+  deploy:
+    # if: github.ref == 'refs/heads/main'
+    needs: [build, docker-build]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: superfly/flyctl-actions/setup-flyctl@master
+      - run: flyctl machine update ${{ secrets.FLY_MACHINE_ID }} --image ghcr.io/filecoin-station/fil-deal-ingester:$(git rev-parse HEAD) 
+        env:
+          FLY_API_TOKEN: ${{ secrets.FLY_API_TOKEN }}
+      - if: failure()
```

Comment:
Please disable Slack notifications for failed deployments while you are debugging this workflow.